File: my-shop-selenium/src/test/java/pages/LoginPage.java
package pages;

import org.openqa.selenium.By;
import org.openqa.selenium.WebElement;

import java.util.List;

public class LoginPage extends BasePage {

    //region Locators
    private static final By emailInput = By.id("email");
    private static final By passwordInput = By.id("passwd");
    private static final By signInButton = By.id("SubmitLogin");
    private static final By loginAlertError = By.cssSelector("div.alert.alert-danger");
    //endregion

    //region Methods
    public static void fillEmail(String email) {
        sendKeys(emailInput, email);
    }

    public static void fillPassword(String password) {
        sendKeys(passwordInput, password);
    }

    public static void clickSignIn() {
        click(signInButton);
    }

    public static void accessPage() {
        get("http://www.automationpractice.pl/index.php?controller=authentication&back=my-account");
    }

    public static List<String> getMeessagesLoginAlertError() {
        return getElement(loginAlertError)
                .findElements(By.tagName("li")).stream()
                .map(WebElement::getText)
                .toList();
    }
    //endregion

    //region Fluxos
    public static void login(String email, String password) {
        accessPage();
        fillEmail(email);
        fillPassword(password);
        clickSignIn();
    }
    //endregion
}

File: my-shop-cypress/cypress/e2e/pages/login-page.ts

export default class LoginPage {

    // Locators
    static emailInput = '#email';
    static passwordInput = '#passwd';
    static signInButton = '#SubmitLogin';
    static loginAlertError = 'div.alert.alert-danger li';
    static pageUrl = Cypress.config("baseUrl") + "?controller=authentication&back=my-account";

    // Methods
    public static fillEmail(email: string): void {
        cy.get(this.emailInput).type(email);
    }

    public static fillPassword(password: string): void {
        cy.get(this.passwordInput).type(password);
    }

    public static clickSignIn(): void {
        cy.get(this.signInButton).click();
    }

    public static accessPage(): void {
        cy.visit(this.pageUrl);
    }

    public static getMeessagesLoginAlertError()  {
        return cy.get(this.loginAlertError);
    }

    // Fluxos
    public static login(email: string, password: string): void {
        this.fillEmail(email);
        this.fillPassword(password);
        this.clickSignIn();
    }

}
